src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -1145,6 +1175,23 @@ private MethodSymbol PerformForEachPatternOverloadResolution(TypeSymbol patternT
                     // We don't want to allow this case.
                     result = null;
```

Comment:
>result = null; [](start = 20, length = 14)

It feels like this is a premature optimization, we probably should still make sure that we can build default arguments if they are needed. Also, external consumers might still be interested in getting information about this method. Probably should follow up separately.

Reply:
Filed [https://github.com/dotnet/roslyn/issues/50164](https://github.com/dotnet/roslyn/issues/50164) to follow up.

---
In reply to: [549831554](https://github.com/dotnet/roslyn/pull/50105#discussion_r549831554) [](ancestors = 549831554)